extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/v2Commands.scala

```diff
@@ -0,0 +1,346 @@
+
+  // other table commands
+
+  val DropTable: V2Command = V2Command(
+    operType = DROPTABLE,
+    buildOutput = (plan, outputObjs, _, _) => {
+      val tableIdent =
+        if (isSparkVersionAtLeast("3.1")) {
+          val resolvedTable = getFieldVal[LogicalPlan](plan, "child")
+          getFieldVal[Identifier](resolvedTable, "identifier")
+        } else {
+          getFieldVal[Identifier](plan, "ident")
+        }
+      outputObjs += v2TablePrivileges(tableIdent)
+    })
+
+  val CacheTable: V2Command = V2Command(
+    operType = CREATEVIEW,
+    leastVer = "3.2",
+    buildInput = (plan, inputObjs, _) => {
+      val query = getFieldVal[LogicalPlan](plan, "table") // table to cache
+      buildQuery(query, inputObjs)
+    })
+
+  val CacheTableAsSelect: V2Command = V2Command(
+    operType = CREATEVIEW,
+    leastVer = "3.2",
+    buildInput = (plan, inputObjs, _) => {
+      val query = getFieldVal[LogicalPlan](plan, "plan")
+      buildQuery(query, inputObjs)
+    })
+
+  val CommentOnNamespace: V2Command = V2Command(
+    operType = ALTERDATABASE,
+    buildOutput = (plan, outputObjs, _, _) => {
+      val resolvedNamespace = getFieldVal[AnyRef](plan, "child")
+      val namespace = getFieldVal[Seq[String]](resolvedNamespace, "namespace")
+      outputObjs += databasePrivileges(quote(namespace))
+    })
+
+  val CommentOnTable: V2Command = V2Command(
+    operType = ALTERTABLE_PROPERTIES,
+    cmdTypes = Seq(
+      if (isSparkVersionAtLeast("3.2")) V2AlterTableCommand else V2DdlTableCommand))
+
+  val MergeIntoTable: V2Command = V2Command(
+    buildInput = (plan, inputObjs, _) => {
+      val table = getFieldVal[DataSourceV2Relation](plan, "sourceTable")
+      buildQuery(table, inputObjs)
+    },
+    buildOutput = (plan, outputObjs, _, _) => {
+      val table = getFieldVal[DataSourceV2Relation](plan, "targetTable")
+      if (table.identifier.isDefined) {
+        outputObjs += v2TablePrivileges(
+          table.identifier.get,
+          actionType = PrivilegeObjectActionType.UPDATE)
+      }
+    })
+
+  val RepairTable: V2Command = V2Command(
+    operType = ALTERTABLE_ADDPARTS,
+    leastVer = "3.2",
+    cmdTypes = Seq(V2DdlTableCommand))
+
+  val TruncateTable: V2Command = V2Command(
+    leastVer = "3.2",
+    buildOutput = (plan, outputObjs, _, _) => {
+      val table = getFieldVal[Any](plan, "table")
+      val tableIdent = getFieldVal[Identifier](table, "identifier")
+      outputObjs += v2TablePrivileges(tableIdent, actionType = PrivilegeObjectActionType.UPDATE)
+    })
+
+  // with V2AlterTableCommand
+
+  val AlterTable: V2Command = V2Command(
+    operType = ALTERTABLE_ADDCOLS,
+    leastVer = "3.0",
+    mostVer = "3.1",
+    buildOutput = (plan, outputObjs, _, _) => {
+      val table = getFieldVal[Any](plan, "table")
+      val tableIdent = getFieldVal[Option[Identifier]](table, "identifier")
+      if (tableIdent.isDefined) {
+        outputObjs += v2TablePrivileges(tableIdent.get)
+      }
```

Comment:
can we adjust the order? here and other related places

Reply:
changed field order of V2Command.